.github/workflows/auto-maintenance.yml

```diff
@@ -64,3 +73,11 @@ jobs:
           token: ${{ secrets.CODECOV_TOKEN }}
           files: coverage/lcov.info
           flags: unittests
+
+      - name: Commit notices 📝
+        if: ${{ github.event_name == 'push' }}
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: Update NOTICES
+          file_pattern: NOTICES
+          branch: ${{ steps.extract_branch.outputs.branch }}
```

Comment:
Is this only committed when the licenses change or after each push with an occasionally empty change log?

Reply:
Do you think it would be better to run this only on merge or what are your thoughts about it?

Reply:
This is just like doing:
```bash
git add .
git commit -m “test”
git push
```
in non-change repo -> Noop -> This will only commit when we change the pubspec.yaml

Reply:
Was just asking because git has `--allow-empty` which would generate a commit message at each push, which seems a little messy to me. But if it only commits on changes to `pubsepc` it's fine :)

Reply:
Agree, you would have to specify that as an extra option with that action.